tests/testthat/test-module_teal.R

```diff
@@ -171,7 +171,7 @@ testthat::describe("srv_teal arguments", {
 testthat::describe("srv_teal teal_modules", {
   testthat::it("are not called by default", {
     shiny::testServer(
-      app = srv_teal,
+      app = teal:::srv_teal_body,
```

Comment:
Just saying - we don't test private functions. We test their effects only through public interface

Reply:
That's what I initially had but apparently the test are failing. See: https://github.com/insightsengineering/teal/actions/runs/10419973058/job/28859103688

The public `srv_teal` does not have access to variables defined in its child and private modules.

Reply:
Testing private function `teal:::srv_teal_body` needs better reasoning that public function didn't have access to some objects. PR won't be accepted until good reason will be provided and test files reorganized. My idea is that everything should be tested through public functions as testing private doesn't guarantee the precision (private functions in tests might be run in a different context that they are executed in the framework, thus their test result are not 100% reliable)